apps/src/labs/labRedux.ts

```diff
@@ -233,12 +225,16 @@ function setProjectAndLevelData(
   if (aborted) {
     return;
   }
-  const {channel, source, levelData} = data;
+  const {channel, sources, levelData} = data;
   dispatch(setChannel(channel));
-  dispatch(setSource(source));
+
+  if (sources) {
```

Comment:
if there aren't sources, should we set sources to undefined?

Reply:
Oh yes good call, don't want old sources to hang around.